zenoh-flow/src/model/mod.rs

```diff
@@ -12,6 +12,43 @@
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
 
+use std::{path::PathBuf, str::FromStr};
+
+use crate::{bail, prelude::ErrorKind, zfresult::ZFError};
+
+use self::registry::NodeKind;
+
 pub mod descriptor;
 pub mod record;
 pub mod registry;
+
+/// The middleware used for the builtin sources and sink.
+#[derive(Debug)]
+pub(crate) enum Middleware {
+    Zenoh,
+}
+
+impl FromStr for Middleware {
+    type Err = ZFError;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let s = s.to_lowercase();
+        if s == "zenoh" {
+            return Ok(Self::Zenoh);
+        }
+        bail!(ErrorKind::ParsingError, "{s} is not a valid middleware!")
+    }
+}
+
+impl ToString for Middleware {
+    fn to_string(&self) -> String {
+        "zenoh".to_string()
+    }
+}
+
+#[derive(Debug)]
+/// Zenoh-Flow's custom URI struct used for loading nodes.
+pub(crate) enum URIStruct {
+    File(PathBuf),
+    Builtin(Middleware, NodeKind),
+}
```

Comment:
I'd suggest the following:

```suggestion
/// The URI supported by Zenoh-Flow.
pub(crate) enum Uri {  // or possibly ZFUri
    File(PathBuf),
    Builtin(Middleware),
}
```

1. Calling an enum `Struct` feels weird; if we need to separate between Zenoh-Flow's URI and the uri from the package we can qualify the path.
2. The `NodeKind` is redundant with the declaration and makes our code heavier (error cases to handle).